ambry-store/src/test/java/com.github.ambry.store/IndexTest.java

```diff
@@ -727,6 +737,80 @@ public void findDeletedEntriesSinceTest() throws InterruptedException, IOExcepti
   }
 
   /**
+   * Tests the index segment roll over when there is a change in IndexValue size. With introduction of
+   * {@link PersistentIndex#VERSION_1} there is a change in IndexValue
+   * @throws StoreException
+   * @throws IOException
+   */
+  @Test
+  public void testIndexSegmentRollOverNewIndexSegmentVersion() throws StoreException, IOException {
+    indexSegmentRollOverTest(true);
+    indexSegmentRollOverTest(false);
+  }
+
+  /**
+   * Tests that the index segment rolls over when there is a version change in the index value or index segment
+   * @param rollOverWithPutRecord {@code true} if index segment roll over is expected with a put record,
+   *                              {@code false} otherwise
+   * @throws StoreException
+   * @throws IOException
+   */
+  private void indexSegmentRollOverTest(boolean rollOverWithPutRecord) throws StoreException, IOException {
+    state.closeAndClearIndex();
+    Offset currentEndOffset = state.index.getCurrentEndOffset();
+    state.index.close();
+
+    ArrayList<IndexEntry> indexEntries = new ArrayList<>();
+    // create an index entry in Version_0
+    IndexEntry entry = new IndexEntry(state.getUniqueId(),
+        IndexValueTest.getIndexValue(CuratedLogIndexState.PUT_RECORD_SIZE, currentEndOffset, state.time.seconds(),
+            PersistentIndex.VERSION_0));
+    // create Index Segment in PersistentIndex.Version_0
+    IndexSegment indexSegment = generateIndexSegmentV0(entry.getValue().getOffset(), entry.getKey().sizeInBytes(),
+        entry.getValue().getBytes().capacity());
+    state.appendToLog(CuratedLogIndexState.PUT_RECORD_SIZE);
+    FileSpan fileSpan = state.log.getFileSpanForMessage(currentEndOffset, CuratedLogIndexState.PUT_RECORD_SIZE);
+    indexSegment.addEntry(entry, fileSpan.getEndOffset());
```

Comment:
In order to create an index segment (segment in version 0 in this case), we need the start offset, key size and value size. Hence we need to create an IndexEntry which will be used to create the indexsegment. Since index entry is already available, its been added explicitly to he segment. Any further puts can use the addPutEntries()